iceoryx_dust/test/moduletests/test_cxx_string_conversion.cpp

```diff
@@ -108,4 +110,618 @@ TYPED_TEST(StdString_test, StringOfSizeCapaToSTDStringConvResultsInSizeCapa)
+    EXPECT_THAT(sut.compare(foo), Gt(0));
+}
+
+/// @note constexpr int64_t compare(const std::string& other) const noexcept
+TYPED_TEST(StdString_test, CompareWithStdStringResultNegativeWithDifferentSize)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "fbfa1376-8474-4cee-8c83-5adb6dc115a8");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    std::string testString(STRINGCAP, 'M');
+    ASSERT_THAT(this->testSubject.unsafe_assign(testString.c_str()), Eq(true));
+    std::string foo(STRINGCAP + 4, 'M');
+    EXPECT_THAT(this->testSubject.compare(foo), Lt(0));
+}
+
+/// @note template <uint64_t Capacity>
+/// inline std::ostream& operator<<(std::ostream& stream, const string<Capacity>& str)
+TYPED_TEST(StdString_test, EmptyStreamInputWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "88c68194-9a9c-4f2f-a0e0-90bd72f9b102");
+    std::ostringstream testStream;
+    testStream << "";
+    EXPECT_THAT(testStream.str(), StrEq(""));
+}
+
+TYPED_TEST(StdString_test, StreamInputOfSizeCapacityWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "c9b3dff3-008d-4189-818f-3534767e7ee4");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    std::string testString(STRINGCAP, 'M');
+    string<STRINGCAP> testFixedString(TruncateToCapacity, testString.c_str(), testString.size());
+    std::ostringstream testStream;
+    testStream << testFixedString;
+    EXPECT_THAT(testStream.str(), Eq(testFixedString.c_str()));
+}
+
+/// @note int64_t compare(const T& other) const noexcept
+/// with T = {std::string}
+TYPED_TEST(StdString_test, CompareEqStdStringResultsInZero)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "b9e8ba17-5de8-4a23-b310-e98c1fb8ecb9");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    std::string testStdString(STRINGCAP, 'M');
+    ASSERT_THAT(this->testSubject.unsafe_assign(testStdString.c_str()), Eq(true));
+
+    EXPECT_THAT(this->testSubject.compare(testStdString), Eq(0));
+}
+
+TYPED_TEST(StdString_test, CompareWithStdStringResultNegative)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "3448a602-886a-4857-b5c3-d216bf01c004");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    std::string temp(STRINGCAP, 'L');
+    ASSERT_THAT(this->testSubject.unsafe_assign(temp.c_str()), Eq(true));
+
+    // required to verify string literal functionality of iox::string
+    // NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
+    char testCharArray[STRINGCAP + 1U];
+    for (uint64_t i = 0U; i < STRINGCAP; ++i)
+    {
+        // NOLINTJUSTIFICATION no other way to populate testCharArray
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+        testCharArray[i] = 'M';
+    }
+
+    const std::string testStdString = &testCharArray[0];
```

Comment:
Seems like this was necessary for the "char array part" of the test and can be now replaced with  `const std::string testStdString(STRINGCAP, 'M');`